server/src/app.ts

```
@@ -40,6 +41,32 @@ app.use(function (req, res, next) {
   next()
 })
 
+app.use(function (req, res, next) {
+  // Exclude createReview and deleteReview from auto-migrations
+  const re = new RegExp('/reviews/[0-9]*');
+
+  if (re.test(req.path)) {
+    return next()
+  }
+
+  // Extract review ID from request
+  let reviewId = req.query.review_id || req.body.review_id || _.last(req.url.split('/'))
+  reviewId = parseInt(reviewId)
+
+  if (reviewId) {
+    adapter.handleMigrations(reviewId)
+    .then(function() {
+      next()
+    })
+    .catch(function (err) {
+      console.log(err)
+      res.status(500).json({ error: err.message })
```

I am trying to think, is it safe to ignore the migration error and move on to the request as a best-effort approach? In some (most?) cases, the request will succeed, and if it fails, it was destined to fail anyway.

---

My worry with this is that it might lead to some data-integrity issues. I think it's best for migrations to fail fast and hard.

---

That might be true. But I cannot overlook the situation where we would have a bug in migrations that will fail the majority of requests where they would normally succeed without attempting migrations.

---

Isn't that what tests and staging is for? :)

---

Yes, but only if we guarantee bugs do not show up later in some edge cases!

---

Agreed to fail early.